CRM/Event/Form/Participant.php

```diff
@@ -733,7 +733,7 @@ public static function formRule($values, $files, $self) {
       if (empty($values['payment_instrument_id'])) {
         $errorMsg['payment_instrument_id'] = ts('Payment Method is a required field.');
       }
-      if ($self->getPriceSetID()) {
+      if ($values['priceSetId']) {
         CRM_Price_BAO_PriceField::priceSetValidation($self->getPriceSetID(), $values, $errorMsg);
```

Comment:
Line 737 also uses `$self->getPriceSetID()` ?

Reply:
@demeritcowboy yeah - I think I thought that was OK - but I've changed it now